### ISSUE TITLE
copying multiple lines from the viewer, the blanks (between the last word of one line and the first word of the following line) will be erased.

### ISSUE THREAD

**Issue by lvyq800** [mozilla/pdf.js#10110]
Attach (recommended) or Link to PDF file here:

Configuration:
- Web browser and its version: any
- Operating system and its version: any
- PDF.js version: the latest
- Is a browser extension: no

Steps to reproduce the problem:
1. select multiple lines of the content from the viewer and copy it.
2. then paste the copy to another place such as notepad
   
![image](https://user-images.githubusercontent.com/27040239/46177393-ca890e80-c306-11e8-8b3c-2d038bde8f79.png)

What is the expected behavior? (add screenshot)
Expected result: 
Dynamic languages such as JavaScript are more difficult to com-pile than statically typed ones. Since no concrete type information is available, traditional compilers need to emit generic code that can

What went wrong? (add screenshot)
These are what I can get:
Dynamic languages such as JavaScript are more difficult to com-pile than statically typed ones. Since no concrete type informationis available, traditional compilers need to emit generic code that can

correct aspect: com-pile is correctly connected. It is better if "com-pile" can be transfered to "compile"
problem: the blank between "information" and "is" is erased. (both words are in different lines)

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):
https://mozilla.github.io/pdf.js/web/viewer.html

**Comment by kfir91** [comment]
same problem

**Comment by luiscastro193** [comment]
This happens because sometimes text selection detects more line breaks than it should. To "fix it", the renderer uses span elements instead of div elements. But that workaround causes this bug.

**Comment by kfir91** [comment]
@luiscastro193  ok, and how can we fix this bug ?

**Comment by luiscastro193** [comment]
Looks like this function should be fixed somewhere:

https://github.com/mozilla/pdf.js/blob/e4d2a1604efdb4a49f29b4a593baea23e4470c36/src/core/evaluator.js#L1276

**Comment by kfir91** [comment]
@luiscastro193 tnx, how i need to change this function?

**Comment by luiscastro193** [comment]
Well the related param is `combineTextItems` (true by default) so apparently one of these if-statements is resolving less often that it should:

https://github.com/mozilla/pdf.js/blob/e4d2a1604efdb4a49f29b4a593baea23e4470c36/src/core/evaluator.js#L1588

https://github.com/mozilla/pdf.js/blob/e4d2a1604efdb4a49f29b4a593baea23e4470c36/src/core/evaluator.js#L1621

No idea which one or why.

**Comment by kfir91** [comment]
@luiscastro193  i changed it to false nothing changed.

**Comment by luiscastro193** [comment]
Changing it to `false` is even worse. It has to be `true`, it's just not working as it should because the function I mentioned has a bug somewhere.

**Comment by kfir91** [comment]
no fix yet?

**Comment by kfir91** [comment]
There is any solving ?

**Comment by timvandermeij** [comment]
No, this is a known issue that someone needs to look into.

**Comment by wesinator** [comment]
also #10640

**Comment by kfir91** [comment]
> No, this is a known issue that someone needs to look into.

Who is the someone?

**Comment by ADrejta** [comment]
For any one that might want to get the text from the a pdf programmatically a temporary "workaround" (that still isn't perfect)
```
const getPageText = async (pdf, pageNo) => {
  const page = await pdf.getPage(pageNo);
  const tokenizedText = await page.getTextContent();
  const pageText = tokenizedText.items.map(token => token.str).join(" "); // this space can sometimes not be the best solution that's why it's still not perfect
  return pageText;
};

const getPDFText = async (source)=> {
  const pdf = await PDFJS.getDocument(source).promise;
  const maxPages = pdf.numPages;
  const pageTextPromises = [];
  for (let pageNo = 1; pageNo <= maxPages; pageNo += 1) {
    pageTextPromises.push(getPageText(pdf, pageNo));
  }
  const pageTexts = await Promise.all(pageTextPromises);
  return pageTexts.join(" ");
};
```
This works with pdf.js versions 2.0+

**Comment by marco-c** [comment]
Could this possibly have been fixed by #13257?

**Comment by Snuffleupagus** [comment]
Duplicate of #9714
